### PR TITLE
feat(npg-apm): api keys configuration for APM

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-infra/pull/1889
Depends on https://github.com/pagopa/pagopa-ecommerce-local/pull/157
Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/188

#### List of Changes
Add api keys configuration for NPG alternative payment methods (APM).
This PR integrate eCommerce commons NpgApiKeyHandler utility class to handle all NPG psp specific api keys

#### Motivation and Context
Those modifications are required in order to use the proper api key based on selected payment methods and psp id

#### How Has This Been Tested?
Junit tests + local tests with eCommerce local

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .devops/code-review-pipelines.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ resources:
     - repository: pagopaEcommerceLocal
       type: github
       name: pagopa/pagopa-ecommerce-local 
-      ref: main
+      ref: CHK-2684-apm-api-keys
```
Comment (pietro-tota) [inline .devops/code-review-pipelines.yml:15]:
```suggestion
      ref: main
```
To be restored after ecommerce local pr merge

File: pom.xml [hunk 2]
```diff
@@ -510,6 +510,8 @@
                     <connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
                     <scmVersionType>tag</scmVersionType>
                     <scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+                    <scmVersionType>branch</scmVersionType>
+                    <scmVersion>CHK-2684-npg-apm-api-keys</scmVersion>
```
Comment (pietro-tota) [inline pom.xml:514]:
```suggestion
```
To be removed after ecommerce commons pr merge

File: src/main/java/it/pagopa/transactions/configurations/NpgPspApiKeysConfigBuilder.java [hunk 3]
```diff
@@ -38,11 +38,114 @@ public it.pagopa.ecommerce.commons.utils.NpgPspApiKeysConfig npgCardsApiKeys(
...
+    ) {
+        return parseApiKeysMap(
+                apiKeys,
+                pspToHandle,
+                NpgClient.PaymentMethod.BANCOMATPAY
+        );
+    }
+
+    @Bean
+    public NpgApiKeyConfiguration npgApiKeyHandler(
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/configurations/NpgPspApiKeysConfigBuilder.java:116]:
```suggestion
    public NpgApiKeyConfiguration npgApiKeyConfiguration(
```

**Review by giovanniberti — APPROVED** [review]

File: src/main/java/it/pagopa/transactions/configurations/NpgPspApiKeysConfigBuilder.java (hunk 3 above)
Comment (pietro-tota) [reply src/main/java/it/pagopa/transactions/configurations/NpgPspApiKeysConfigBuilder.java:116]: ops, my bad, missing during refactoring :D

Comment (pietro-tota) [reply src/main/java/it/pagopa/transactions/configurations/NpgPspApiKeysConfigBuilder.java]: Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/465/commits/00ef5f6eb9707df8bf2234c1445436559e23b985

**Review by giovanniberti — APPROVED** [review]

**Review by giovanniberti — APPROVED** [review]

**Review by ciuffagianluca — APPROVED** [review]

**Pull request merged**